### PR TITLE
Add null checks, other misc fixes

### PR DESCRIPTION
**Pull request by tinker-michaelj**
**Description**:
 - [Logs at `WARN` and skip](https://github.com/hashgraph/hedera-services/blob/fix-migration-npes/hedera-node/src/main/java/com/hedera/services/state/migration/KvPairIterationMigrator.java#L74) any key/value pairs that reference non-existent contracts.
 - Allows contracts to have [null `autoRenewAccount`](https://github.com/hashgraph/hedera-services/blob/fix-migration-npes/hedera-node/src/main/java/com/hedera/services/context/primitives/StateView.java#L557) when answering e.g. `ContractGetInfo`.
 - Avoids a possible NPE in `ContextOptionValidator` precheck on the first transaction submitted to a network after restart.
 - Stops logging at `WARN` in `HTSPrecompiledContract` when a `TransactionBody` cannot be parsed.
 - Rejects `CryptoTransfer`s with a [`tokenTransferList` that repeats NFT serial numbers](https://github.com/hashgraph/hedera-services/blob/fix-migration-npes/hedera-node/src/main/java/com/hedera/services/ledger/PureTransferSemanticChecks.java#L191) (this can be removed in 0.27).
 - Catches precheck exception in `DeleteAllowance` transaction that references an invalid `TokenID`.

### CODE REVIEW

**Review by Neeharika-Sompalli — APPROVED** [review]
LGTM !

File: hedera-node/src/main/java/com/hedera/services/ledger/PureTransferSemanticChecks.java [hunk 1]
```diff
@@ -178,18 +179,23 @@ ResponseCodeEnum validateTokenTransferSemantics(List<TokenTransferList> tokenTra
...
 		final var ownershipChanges = tokenTransfers.getNftTransfersList();
-		for (var ownershipChange : ownershipChanges) {
+		uniqueSerialNos.clear();
+		for (final var ownershipChange : ownershipChanges) {
 			if (ownershipChange.getSenderAccountID().equals(ownershipChange.getReceiverAccountID())) {
 				return ACCOUNT_REPEATED_IN_ACCOUNT_AMOUNTS;
 			}
+			if (!uniqueSerialNos.add(ownershipChange.getSerialNumber())) {
+				return INVALID_ACCOUNT_AMOUNTS;
+			}
```
Comment (Neeharika-Sompalli) [inline hedera-node/src/main/java/com/hedera/services/ledger/PureTransferSemanticChecks.java:198]: Why do we need this check ?

**Review by Neeharika-Sompalli — APPROVED** [review]
Other than a check in `PureTransferSemanticChecks.java` , LGTM

File: hedera-node/src/main/java/com/hedera/services/ledger/PureTransferSemanticChecks.java (hunk 1 above)
Comment (Neeharika-Sompalli) [reply hedera-node/src/main/java/com/hedera/services/ledger/PureTransferSemanticChecks.java:198]: Discussed its for mirror node compatibility

**Review by Neeharika-Sompalli — DISMISSED** [review]

**Review by Neeharika-Sompalli — APPROVED** [review]
LGTM !

**Review by anighanta — APPROVED** [review]

**Pull request merged**